mllib/src/main/scala/org/apache/spark/ml/optim/WeightedLeastSquares.scala

```diff
@@ -439,8 +439,9 @@ private[ml] object WeightedLeastSquares {
 
     /**
      * Weighted population standard deviation of labels.
+     * We prevent variance from negative value caused by numerical error.
      */
-    def bStd: Double = math.sqrt(bbSum / wSum - bBar * bBar)
+    def bStd: Double = math.sqrt(math.max(bbSum / wSum - bBar * bBar, 0.0))
```

Comment:
There are a couple more places where variance is computed in this file -- I think they need this too?